tests/unit/lms/services/launch_verifier_test.py

```diff
@@ -169,48 +173,71 @@ def launch_verifier(self, pyramid_request):
 
     @pytest.fixture
     def pyramid_request(self, pyramid_request):
-        # `pyramid.testing.DummyRequest` sets `url` and `path_url` attrs without
-        # a path by default. oauthlib however normalizes the URL to include a
-        # path when it generates the signature base string, whereas PyLTI does
-        # not. Add an empty path to both to match a real application.
-        pyramid_request.url += "/"
-        pyramid_request.path_url += "/"
-
-        # Add the OAuth 1 params (version, nonce, timestamp, ...)
-        oauthlib_client = oauthlib.oauth1.Client(
-            pyramid_request.params["oauth_consumer_key"]
-        )
-        oauthlib_request = oauthlib.common.Request(
-            pyramid_request.url, pyramid_request.method
-        )
-        pyramid_request.params = dict(
-            oauthlib_client.get_oauth_params(oauthlib_request)
-        )
```

Comment:
We were doing very similar things twice in this and the `sign` method, and they both need to be different now, so I've separated all the OAuth1 stuff into a class we use as a fixture to get this stuff done.